src/tree.js

```diff
@@ -83,6 +84,11 @@ export function createInternal(vnode, parentInternal) {
 	return internal;
 }
 
+const shouldSearchComponent = internal =>
+	internal._flags & TYPE_COMPONENT &&
+	(!(internal._flags & TYPE_ROOT) ||
+		internal.props._parentDom == getParentDom(internal._parent));
```

Comment:
This is the logic that says only search through a Component if it is not a portal/root node or if it is a root node, search through it only if it's parentDom matches its parent's parentDom